src/Test/TestCombat.java

```diff
@@ -54,6 +58,7 @@ private boolean testUnitDeath()
 
     // Clean up
     testMap.removeUnit(mechA);
+    CombatEngine.modifiers.remove(mod);
```

Comment:
Oh, but it is necessary in normal gameplay. You just put `MapController` in charge of it.